iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -608,7 +618,7 @@ class string
 
     // safe access is guaranteed since the char array is wrapped inside the string class
     // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
-    char m_rawstring[Capacity + 1U]{'\0'};
+    char m_rawstring[Capacity + 1U]{};
```

Comment:
Ok, empty initializer list initializes it to `\0` as well.

@FerdinandSpitzschnueffler I am fairly certain that we can safe initialization effort by only initializing in the ctor (i.e. the actual data followed by a `\0`.). Ideally this would be the case, as 0 initialization is not free and should be avoided if not needed.

Nothing for now, but just to maybe consider in some future refactoring.